src/main/java/unicash/model/transaction/Name.java

```diff
@@ -10,13 +10,14 @@
 public class Name {
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Names should only contain alphanumeric characters and spaces, and it should not be blank";
+            "Names should only contain alphanumeric characters and spaces up to 500 characters, "
+                    + "and it should not be blank";
 
     /*
      * The first character of the transaction must not be a whitespace,
      * otherwise " " (a blank string) becomes a valid input.
      */
-    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+    public static final String VALIDATION_REGEX = "^[\\p{Alnum}][\\p{Alnum} ]{0,499}$";
```

Comment:
I think the validation regex for names can be made as lenient as location information! I don't see a reason why we cannot support non-alnum characters in names